lib/parser/parser-utils.ts

```diff
@@ -150,3 +210,103 @@ function findComponentPropertyTemplate(
 function isPropertyObject(property: any): property is ComponentProperty {
     return property instanceof Object;
 }
+
+/**
+ * Builds default content for when a component is created.
+ * This goes through the component properties and looks for default values.
+ * These defaultValues are stored in a model object, that's used as initial
+ * data for the component when inserted.
+ *
+ * Default values are only supported for the basic data types (styles, inlineStyles and data).
+ * Directive default values are excluded for now, as in general they are not needed.
+ * In case we do want to expand support, consider trying to merge the logic of creating default
+ * content with the properties classes, as both deal with the article format.
+ * Alternatively the default values could be set after component creation, but this would
+ * generate more data operations and trigger more view updates (needs to be tested through).
+ *
+ * @param componentSet
+ */
+function buildComponentSetDefaultContent(componentSet: ComponentSet) : void {
+    for (let component of Object.values(componentSet.components)) {
+        buildComponentDefaultContent(componentSet.defaultComponentContent, component);
+    }
+}
+
+/**
+ * Build default component model for given component and add to defaultComponentContent input.
+ *
+ * @param defaultComponentContent
+ * @param component
+ */
+function buildComponentDefaultContent(
+    defaultComponentContent: ComponentSet['defaultComponentContent'],
+    component: ParsedComponent
+) : void {
+    for (let property of component.properties) {
+        buildComponentPropertyDefaultContent(defaultComponentContent, component.name, property);
+    }
+}
+
+/**
+ * Build default component model property data and add to defaultComponentContent.
+ *
+ * @param defaultComponentContent
+ * @param componentName
+ * @param property
+ */
+function buildComponentPropertyDefaultContent(
+    defaultComponentContent: ComponentSet['defaultComponentContent'],
+    componentName: string,
+    property: ComponentProperty,
+) : void {
+    // No default value
+    if (!property.defaultValue) {
+        return;
+    }
+
+    switch (property.dataType) {
```

Comment:
Can you make sure all cases have matching tests? Currently only styles is tested.